### ISSUE TITLE
bug: SimulatedAnnealingSearch shouldAccept method should not instance new Random() on each call.

### ISSUE THREAD

**Issue by tcerda95** [aimacode/aima-java#405]
**Problem**
Each call to `new Random()` instances a new random generator with a distinct seed. Therefore subsequent calls to `new Random().nextDouble()` does not guarantee uniform distribution between 0.0 and 1.0 as desired. Current code ([SimulatedAnnealingSearch.java](https://github.com/aimacode/aima-java/blob/AIMA3e/aima-core/src/main/java/aima/core/search/local/SimulatedAnnealingSearch.java)):

```Java
private boolean shouldAccept(double temperature, double deltaE) {
		return (deltaE > 0.0)
				|| (new Random().nextDouble() <= probabilityOfAcceptance(temperature, deltaE));
}
```

**Solution**
Instance Random a single time in the constructor OR simply use `Math.random()`. This is the preferred solution since there is no need to manage any Random instance and Math uses a unique Random instance as implied by the Java SDK documentation. Solution code:

```Java
private boolean shouldAccept(double temperature, double deltaE) {
		return (deltaE > 0.0)
				|| (Math.random() <= probabilityOfAcceptance(temperature, deltaE));
}
```

**Resources**
 [Random.java](https://docs.oracle.com/javase/7/docs/api/java/util/Random.html)
 [Math.random()](https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#random--)

I can make a PR fixing the issue if necessary.

**Comment by samagra14** [comment]
@tcerdaITBA It is a nice catch. Please go ahead with the Pull Request.

**Comment by tcerda95** [comment]
Fix #406 has been merged.
Closing issue.
Thanks!
